driver/src/main.rs

```diff
@@ -202,13 +212,28 @@ fn main() {
     let price_finder = price_finding::create_price_finder(fee, options.solver_type, price_oracle);
 
     // Create the orderbook reader.
-    let orderbook = PaginatedStableXOrderBookReader::new(&contract, options.auction_data_page_size);
-    info!("Orderbook filter: {:?}", options.orderbook_filter);
+    let primary_orderbook =
+        PaginatedStableXOrderBookReader::new(contract.clone(), options.auction_data_page_size);
+
+    // NOTE: Keep the shadowed orderbook around so it doesn't get dropped and we
+    //   can pass a reference to the filtered orderbook reader.
+    let unfiltered_orderbook: Box<dyn StableXOrderBookReading + Sync> = if options
+        .use_shadowed_orderbook
+    {
+        let shadow_orderbook =
+            PaginatedStableXOrderBookReader::new(contract.clone(), options.auction_data_page_size);
```

Comment:
This order book seems identical to the primary order book (since they are both acquired via the same `PaginatedStableXOrderBookReader`). I thought that each order book was meant to be fetch in a different way so that comparison would serve a purpose.

Reply:
Yes, but we only have one for now (another one is coming in a separate PR). This does verify that the orderbook retrieval does not depend on when in the batch it happens.